### ISSUE TITLE
Calendar: Add Live Region to Header to track programatic changes in date

### ISSUE THREAD

**Issue by harrisonwillettbetterworks** [primefaces/primevue#4074]
### Describe the bug

As documented in Tickets [#3877](https://github.com/primefaces/primevue/issues/3877) and [#3930](https://github.com/primefaces/primevue/issues/3930) there is an insufficient response for screen reader users when using the product.

### Steps to reproduce the behavior for Scenario 1:
_Failure of WCAG SC 4.1.2_
1. Given I am using a screen reader ([Screen Reader](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn/related) - Chrome Plug in is sufficient for testing/dev purposes)
2. And I am on this page [https://primevue.org/calendar/#basic](https://primevue.org/calendar/#basic) 
3. And I tab into the basic calendar input
4. Then I am asked to choose the day but I do not know what my month and year are yet as I can not see
5. And When pressing "tab" OR "shift" + "tab" I get notified of the actions of these buttons but not values being set

### Expected behavior to Scenario 1:
Inside `class="p-datepicker-header"` please add a [live region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) or regions with a class of "p-sr-only" to allow for the announcement of the month and or year when changed. These will need to be set every time the month or year is changed.

### Steps to reproduce the behavior for Scenario 2:
_Failure of WCAG SC 4.1.2_
1. Given I am using a screen reader ([Screen Reader](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn/related) - Chrome Plug in is sufficient for testing/dev purposes)
2. And I am on either of these pages [https://primevue.org/calendar/#monthpicker](https://primevue.org/calendar/#monthpicker) or  [https://primevue.org/calendar/#yearpicker] (https://primevue.org/calendar/#yearpicker)
3. And I tab into the calendar input
4. Once I hit enter or space I should be able to traverse the options within the role of listbox
5. Once on the listed options I do not hear a role, aria-selected state, aria-setsize or aria-posinset

### Reproducer

https://primevue.org/calendar/#basic

### PrimeVue version

3.29.2

### Vue version

3.x

### Language

TypeScript

### Build / Runtime

Vue CLI App

### Browser(s)

Chrome Version 114.0.5735.106
